### ISSUE TITLE
on using `build`

### ISSUE THREAD

**Issue by sdx23** [albertoruiz/hmatrix#194]
`build` doesn't make sense to me. Why is the generating function `(e -> e -> e)` and not `(Int -> Int -> e)`? Is this done to avoid overhead from `fromInteger` calls as someone on #haskell suggested? If so, I'd suggest mentioning this in the haddock.

**Comment by albertoruiz** [comment]
Yes, this is done to avoid `fromIntegral` calls. Some definitions look better but I am not sure it is a good idea.

**Comment by sdx23** [comment]
How I came about this was when implementing `tril` (triangular lower matrix part of a matrix). In such a case I'll be using the argument `e`s as indices and have to `round` them - which seems fairly odd, given that it's indices.

**Comment by dschrempf** [comment]
Hello, I just wanted to open the same issue. I have to round Doubles to get Ints which I use as indices. I think this is really odd. Maybe it makes sense to provide two different ways of /building/ a matrix?

**Comment by albertoruiz** [comment]
I agree, it was a bad idea. It is better to use the actual type of the indices. That function should be fixed.
